libs/live-wallet/src/walletsync/__tests__/trustchainLifecyle.test.ts

```diff
@@ -0,0 +1,84 @@
+import { MockSDK } from "@ledgerhq/trustchain/mockSdk";
+import { setupServer } from "msw/node";
+import { http, HttpResponse } from "msw";
+import { emptyState, convertLocalToDistantState } from "../__mocks__";
+import { trustchainLifecycle } from "..";
+import { setEnv } from "@ledgerhq/live-env";
+
+describe("trustchainLifecycle", () => {
+  let removedCount = 0;
+  let storedData: string | null = null;
+  let storedVersion = 0;
+
+  setEnv("CLOUD_SYNC_API", "http://localhost");
+
+  const handlers = [
+    http.delete("http://localhost/atomic/v1/:slug", ({ request }) => {
+      const authHeader = request.headers.get("Authorization");
+      if (authHeader !== "Bearer old-jwt") {
+        return HttpResponse.json({}, { status: 401 });
+      }
+      removedCount++;
+      storedData = null;
+      storedVersion = 0;
+      return new HttpResponse(null, { status: 204 });
+    }),
+    http.post("http://localhost/atomic/v1/:slug", async ({ request }) => {
+      const authHeader = request.headers.get("Authorization");
+      if (authHeader !== "Bearer new-jwt") {
+        return HttpResponse.json({}, { status: 401 });
+      }
+      const version = parseInt(new URL(request.url).searchParams.get("version") || "0", 10);
+      const json = await request.json();
+      const { payload } = json as { payload: string };
+      storedData = payload;
+      storedVersion = version;
+      return HttpResponse.json({ status: "updated" });
+    }),
+  ];
+
+  let mswServer: ReturnType<typeof setupServer>;
+  beforeAll(() => {
+    mswServer = setupServer(...handlers);
+    mswServer.listen();
+  });
+
+  afterAll(() => {
+    mswServer.close();
+  });
```

Comment:
You can move all of that outside in a specific file, maybe in the future we will have to do same thing on another files? 

wdyt?

Reply:
mmh good point but i think here the code is way too test specific for a generalisation at the moment.
things like

`if (authHeader !== "Bearer new-jwt") {`

are extremely specific for the trustchain rotation test.